### PR TITLE
Add descriptions

### PR DESCRIPTION
**Pull request by krosa**
[#928](https://github.com/sindresorhus/awesome/issues/928)

Added descriptions for Android, React Native, macOS, Swift, Python, C/C++, Java, PHP, CSS, and Kubernetes.

### CODE REVIEW

**Comment by shlomif** [comment]
@krosa: thanks! It looks good except for:

1. See https://github.com/shlomif/Freenode-programming-channel-FAQ/blob/master/FAQ_with_ToC__generated.md#what-is-the-difference-between-scripting-languages-such-as-perl-php-python-or-ruby-and-industrial-strength-languages-such-as-c-c-java-and-c regarding "scripting language".

2. Java, while somewhat resembling C++ in its syntax, has different syntax and semantics and is not compatible with it.

**Comment by krosa** [comment]
Thank you for your input, @shlomif. I edited my Java description to, "Object-oriented language that produces software for multiple platforms."

**Comment by shlomif** [comment]
@krosa : thanks and you are welcome.

**Comment by lauthieb** [comment]
Thanks @krosa! I've approved it.

File: readme.md [hunk 1]
```diff
@@ -148,20 +148,20 @@
...
-- [C/C++](https://github.com/fffaraz/awesome-cpp#readme)
+- [C/C++](https://github.com/fffaraz/awesome-cpp#readme) - General-purpose object-oriented language.
 - [R](https://github.com/qinwf/awesome-R#readme)
 - [D](https://github.com/zhaopuming/awesome-d#readme)
 - [Common Lisp](https://github.com/CodyReichert/awesome-cl#readme)
 - [Perl](https://github.com/hachiojipm/awesome-perl#readme)
 - [Groovy](https://github.com/kdabir/awesome-groovy#readme)
 - [Dart](https://github.com/yissachar/awesome-dart#readme)
-- [Java](https://github.com/akullpp/awesome-java#readme)
+- [Java](https://github.com/akullpp/awesome-java#readme) - Object-oriented language that produces software for multiple platforms.
```
Comment (sindresorhus) [inline readme.md]: > that produces software for multiple platforms.

Almost all languages do this. It's not a useful description.

File: readme.md [hunk 2]
```diff
@@ -127,10 +127,10 @@
 	- [Observables](https://github.com/sindresorhus/awesome-observables#readme)
 	- [npm scripts](https://github.com/RyanZim/awesome-npm-scripts#readme) - Task runner.
 	- [30 Seconds of Code](https://github.com/30-seconds/30-seconds-of-code#readme) - Code snippets you can understand in 30 seconds.
-- [Swift](https://github.com/matteocrippa/awesome-swift#readme)
+- [Swift](https://github.com/matteocrippa/awesome-swift#readme) -  Compiled programming language for iOS, macOS, watchOS, tvOS, and Linux applications.
```
Comment (sindresorhus) [inline readme.md]: This should rather describe what Swift is than mention where it runs, as it also runs on Windows now and more in the future.

File: readme.md [hunk 3]
```diff
@@ -148,20 +148,20 @@
 - [Julia](https://github.com/svaksha/Julia.jl#readme)
 - [Lua](https://github.com/LewisJEllis/awesome-lua#readme)
 - [C](https://github.com/aleksandar-todorovic/awesome-c#readme)
-- [C/C++](https://github.com/fffaraz/awesome-cpp#readme)
+- [C/C++](https://github.com/fffaraz/awesome-cpp#readme) - General-purpose object-oriented language.
```
Comment (sindresorhus) [inline readme.md]: This should describe what's unique about it.

**Comment by krosa** [comment]
@sindresorhus, I edited my descriptions for Java, Swift, and C/C++. Thank you for your input. I'll continue improving as required.

File: readme.md [hunk 4]
```diff
@@ -128,10 +128,10 @@
 	- [Observables](https://github.com/sindresorhus/awesome-observables#readme)
 	- [npm scripts](https://github.com/RyanZim/awesome-npm-scripts#readme) - Task runner.
 	- [30 Seconds of Code](https://github.com/30-seconds/30-seconds-of-code#readme) - Code snippets you can understand in 30 seconds.
-- [Swift](https://github.com/matteocrippa/awesome-swift#readme)
+- [Swift](https://github.com/matteocrippa/awesome-swift#readme) -  Compiled programming language widely used for developing secure and fast apps for iOS.
```
Comment (sindresorhus) [inline readme.md]: Again, this is not describing the language itself.

File: readme.md [hunk 5]
```diff
@@ -149,20 +149,20 @@
...
-- [C/C++](https://github.com/fffaraz/awesome-cpp#readme)
+- [C/C++](https://github.com/fffaraz/awesome-cpp#readme) - General-purpose language best for projects focused on low-level processing and fast applications for server-side software.
 - [R](https://github.com/qinwf/awesome-R#readme)
 - [D](https://github.com/zhaopuming/awesome-d#readme)
 - [Common Lisp](https://github.com/CodyReichert/awesome-cl#readme)
 - [Perl](https://github.com/hachiojipm/awesome-perl#readme)
 - [Groovy](https://github.com/kdabir/awesome-groovy#readme)
 - [Dart](https://github.com/yissachar/awesome-dart#readme)
-- [Java](https://github.com/akullpp/awesome-java#readme)
+- [Java](https://github.com/akullpp/awesome-java#readme) - Secure and platform-independent object-oriented language.
```
Comment (sindresorhus) [inline readme.md]: My previous comment https://github.com/sindresorhus/awesome/pull/1567#discussion_r285533778 still applies to the new description.

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**